pkg/controller/manifestwork/manifestwork_controller.go

```diff
@@ -208,34 +205,31 @@ func (r *ReconcileManifestWork) deleteManifestWorks(
 	}
 
 	// only have klusterlet manifest works, delete klusterlet manifest works
-	klusterletName := fmt.Sprintf("%s-%s", cluster.Name, klusterletSuffix)
+	klusterletName := fmt.Sprintf("%s-%s", cluster.Name, constants.KlusterletSuffix)
 	klusterletWork := &workv1.ManifestWork{}
 	err = r.clientHolder.RuntimeClient.Get(ctx, types.NamespacedName{Namespace: cluster.Name, Name: klusterletName}, klusterletWork)
 	if errors.IsNotFound(err) {
 		// the klusterlet work could be deleted, ensure the klusterlet crds work is deleted
 		return reconcile.Result{}, helpers.ForceDeleteManifestWork(ctx, r.clientHolder.RuntimeClient, r.recorder,
-			cluster.Name, fmt.Sprintf("%s-%s", cluster.Name, klusterletCRDsSuffix))
+			cluster.Name, fmt.Sprintf("%s-%s", cluster.Name, constants.KlusterletCRDsSuffix))
 	}
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// if the manifest work is not applied, we do nothing to avoid to delete the cluster prematurely
-	// Note: there is a corner case, the registration-agent is availabel, but the work-agent is unavailablel,
-	// this will cause that the klusterlet work cannot be deleted, we need user to handle this manually
-	if !meta.IsStatusConditionTrue(klusterletWork.Status.Conditions, workv1.WorkApplied) {
-		log.Info(fmt.Sprintf("delete the manifest work %s until it is applied ...", klusterletWork.Name))
-		return reconcile.Result{}, nil
-	}
+	// Note: we don't wait for the manifest work is applied, so there is a corner case: when the cluster is availabel
+	// but the klusterlet works is not applied, in this time, user delete the cluster, this will cause that the
+	// klusterlet cannot be deleted from the mangaed cluser, we need user to handle this manually
 
 	return reconcile.Result{}, helpers.DeleteManifestWork(
 		ctx, r.clientHolder.RuntimeClient, r.recorder, klusterletWork.Namespace, klusterletWork.Name)
 }
 
 func createKlusterletCRDsManifestWork(managedCluster *clusterv1.ManagedCluster, importSecret *corev1.Secret) *workv1.ManifestWork {
 	crdsKey := constants.ImportSecretCRDSV1YamlKey
-	if !helpers.IsAPIExtensionV1Supported(managedCluster.Status.Version.Kubernetes) {
-		log.Info("crd v1 is not supported, add v1beta1")
+	if managedCluster.Status.Version.Kubernetes != "" &&
```

Comment:
managedCluster.Status.Version.Kubernetes != "" this seems redundant.

Reply:
it's not redundant, because if the version is empty, the  IsAPIExtensionV1Supported will return false